### DISCUSSION TITLE
front-end for devonfw-ide

### DISCUSSION THREAD

**Discussion by hohwille** [devonfw/ide#926] [Ideas]
devonfw-ide is great for power-users. However, new users not familiar with devonfw-ide sometimes have a hard time to setup their own settings git repo and use devonfw-ide properly in their teams according to their requirements and needs. With support we can help those people to get everything correct. However, a UI could support the users to get everything right in a guided and intuitive way.
We already have a prototype of such UI in [devonfw-dashboard](https://github.com/devonfw/dashboard).

Should we revive dashboard and support the user with more guidance?

**Comment by hohwille** [comment]
See also https://github.com/devonfw/dashboard/issues/94#issuecomment-1247938935

**Comment by maybeec** [comment]
I would not recommend a UI honestly, better provide a generator like code.quarkus.io to properly give a starting point and provide good usage information. For example you could have a hybrid page allowing to generate your settings but also giving you just in time the knowledge on selection on how you would have done this on the level of code. Thus, the generator would also be a catalogue of small guides. 

UI always brings in complexity in regards to OS support, testing and so on. I barely doubt that this wil bring the devonfw ide to new users.

Instead let's have a look at a more self explanatory and guiding console interface like the one of jhipster.

**Reply by hohwille** [reply]
> better provide a generator like code.quarkus.io to properly give a starting point

What users say is that they need more than just a starting point (that could be addressed with a website generator) - they want to be informed automatically about new versions available, they want to be guided when they want to change configs like code-style in IDE, etc. 

> UI always brings in complexity in regards to OS support, testing and so on.

IMHO the former dashboard team has solved the hard part to set it up already.
Instead of dumping this invest my suggestion would be to give it a try in building some features into it that bring the most value and see it that pays off.

> I barely doubt that this wil bring the devonfw ide to new users.

We have hundreds of documentation pages with instructions but non-geek-users simply struggle with the complexity.
With new CLI features we now can do things like `devon java version set 11` and hitting [Tab] key will give you tab completion on the matching versions. Building such things in CLIs is sometimes as hard or even harder as building a UI (when having a good foundation for it). However, a UI with a ComboBox prefilled with the current version, an info icon showing if newer version is available and the option to filter and find versions, add tools, etc. will make the life way simpler for end-users.

There are many more advanced UI ideas that I would not build in the first place but just to give a little bit of a vision:
Also an editor for config (`devon.properties` that I do not have to browse on my disc but where I can choose between global, project, and custom config) with TAB-completion would make it perfect. Also assume a view that could show the resolved `devon.properties` with coloring which config comes from global, project, custom config could massively boost debugging.

I would consider myself quite a shell expert. However, looking at CobiGen I always prefer the UI over the CLI.
